ui/src/components/SearchInput/index.vue

```diff
@@ -0,0 +1,185 @@
+  },
+  {
+    key: 'regex' as const,
+    icon: Regex,
+    label: '使用正则表达式',
+    toggleFn: toggleRegex,
+  },
+];
+
+watch(
+  () => searchKey.value,
+  value => {
+    if (!value) {
+      props.searchAddon.clearDecorations();
+      props.searchAddon.clearActiveDecoration();
+    }
+  }
+);
+
+const keyWordsSearch = (value: string) => {
+  if (value) {
+    props.searchAddon.findNext(value, searchOptions);
+  }
+};
+
+const handleSearchPrevious = () => {
+  props.searchAddon.findPrevious(searchKey.value, searchOptions);
+};
+
+const handleSearchNext = () => {
+  props.searchAddon.findNext(searchKey.value, searchOptions);
+};
+
+const closeSearch = (shouldEmit = false) => {
+  searchKey.value = '';
+  props.searchAddon.clearDecorations();
+  props.searchAddon.clearActiveDecoration();
+
+  if (shouldEmit) {
+    emit('close');
+  }
+};
+
+const handleCloseSearch = () => closeSearch(true);
+</script>
+
+<template>
+  <div
+    class="absolute flex items-center top-2 right-2 p-2 z-999 shadow-md rounded-md"
+    :style="{ backgroundColor: lighten(10) }"
+  >
+    <n-flex align="center" class="!gap-2">
+      <n-input
+        v-model:value="searchKey"
+        clearable
+        size="small"
+        :style="{ width: '45%', flex: '1' }"
+        @update:value="keyWordsSearch"
+      />
+
+      <n-button quaternary size="tiny" @click="handleSearchPrevious">
+        <template #icon>
+          <ChevronUp />
+        </template>
+      </n-button>
+      <n-button quaternary size="tiny" @click="handleSearchNext">
+        <template #icon>
+          <ChevronDown />
+        </template>
+      </n-button>
+    </n-flex>
+
+    <n-divider vertical :style="{ height: '24px', backgroundColor: darken(1) }" />
+
+    <n-flex align="center" class="!flex-nowrap !gap-2">
+      <n-flex align="center" class="!flex-nowrap !gap-2">
+        <div
+          v-for="option in searchOptionButtons"
+          :key="option.key"
+          class="search-option-btn"
+          :class="{ active: searchOptions[option.key] }"
+          :style="getActiveStyle(Boolean(searchOptions[option.key]))"
+          :title="option.label"
+          tabindex="0"
+          @click="option.toggleFn"
+        >
+          <component :is="option.icon" color="#fff" :size="16" />
+        </div>
+      </n-flex>
+
+      <n-button quaternary size="tiny" @click="handleCloseSearch">
+        <template #icon>
+          <X />
+        </template>
+      </n-button>
+    </n-flex>
+  </div>
+</template>
+
+<style scoped lang="scss">
+.search-option-btn {
+  padding: 4px 6px;
+  border-radius: 4px;
+  cursor: pointer;
+  transition: all 0.2s ease-in-out;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  user-select: none;
+  background-color: transparent;
+
+  &:hover {
+    background-color: rgba(255, 255, 255, 0.1);
+    transform: scale(1.05);
+  }
+
+  &:active {
+    background-color: rgba(255, 255, 255, 0.2);
+    transform: scale(0.95);
+  }
+}
+</style>
```

Comment:
The provided snippet does not contain significant differences compared to the knowledge cutoff of September 2021 which would be difficult without further context on what is considered irregularity or potential issue. However, generally speaking, this layout structure follows best practices for Vue 3 UI components:

- The Vue framework's `@define*` method allows declarative usage like defining props and emits in templates.

- Reactive state manages dependencies between component interactions through property watches.

However, it would be beneficial with more comprehensive knowledge about its expected functionality and current version constraints before making such an evaluation.